core/src/main/java/io/grpc/inprocess/InProcessTransport.java

```diff
@@ -100,6 +100,7 @@ public void run() {
       shutdownThread.setDaemon(true);
       shutdownThread.setName("grpc-inprocess-shutdown");
       shutdownThread.start();
+      return;
```

Comment:
It's a bug fix found with the tests. Without the return transportReady could be called after transportShutdown and transportTerminated. In addition, it is wrong to call transportReady if we fail to "connect."